strength_percentiles.py

```diff
@@ -176,12 +199,189 @@ def populate_database(webpage, database):
+        result_and_categories = parse_row(row)
         if result_and_categories: # row contains results of meet
             cursor.execute(result_storage_string, result_and_categories)
 
     connection.commit()
     connection.close()
 
+####### Pandas functions
+def find_percentile(dataframe, lifts):
+    """
+    Finds the percentile of the input squat, bench, deadlift and total numbers
+    among the sample of results in the database.
+
+    Parameters:
+    -----------
+    dataframe: pandas dataframe
+        Dataframe containing squat, bench, deadlift, and/or total numbers
+        against which to compare the input numbers.
+    lifts: dictionary with string keys and float (or None) values
+        A dictionary that contains keys ('squat', 'bench', 'deadlift', 'total').
+        The values are floats representing someone's performance in the lift.
+
+    Returns:
+    --------
+    percentile: float
+        The user's percentile rank among the lifters in the database in the
+        input categories.
+
+    """
+    percentiles = []
+    for lift in (SQUAT, BENCH, DEADLIFT, TOTAL):
+        if lifts[lift]: # valid entry for lift
+            entered_lift = lifts[lift]
+            all_competitor_lifts = dataframe[lift]
+            number_of_lifts = all_competitor_lifts.count()
+            number_of_smaller_lifts = all_competitor_lifts[all_competitor_lifts < entered_lift].count()
+            percentile = (float(number_of_smaller_lifts)/number_of_lifts) * 100
+            percentiles.append(lift + ": " + str(percentile) + " percentile")
+        else: # no lift was entered
+            percentiles.append("N/A")
+    return percentiles
+
+def get_population_by_categories(connection, table, categories):
+    """
+    Returns a dataframe containing the powerlifting results of the people who
+    fit the input categories.
+    Parameters:
+    -----------
+    connection: sqlite3 connection object
+        Connection to the database containing desired table
+    table: string
+        string specifying the name of the table containing desired
+        powerlifting meet results
+    categories: dictionary
+        Contains values for gender, professional_status, and equipment
+        fields to use to query the table
+    Returns:
+    --------
+    population_dataframe: pandas dataframe
+        dataframe containing the meet results of all lifters who fit the
+        input criteria.
+
+    """
+    get_population_string = """Select * from {table}
+        WHERE gender LIKE "{gender}%" AND
+        professional_status LIKE "{professional_status}%" and
+        equipment LIKE "{equipment}%";
+        """.format(table=table,**categories)
+    cursor = connection.cursor()
+    population_database = cursor.execute(get_population_string)
+    population_dataframe = pd.read_sql(get_population_string, connection)
+    return population_dataframe
+
+def find_average(dataframe, lift):
```

Comment:
Nit: name the var df instead of dataframe. Not a big deal, but almost every stackoverflow question on pandas has dataframe named as df.

Also, not sure if something like this really needs its own function.